web/src/test/java/org/cbioportal/genome_nexus/web/CancerHotspotsIntegrationTest.java

```diff
@@ -22,18 +27,40 @@
     webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT,
     properties = {
         "vep.url=http://grch37.rest.ensembl.org/vep/human/hgvs/VARIANT?content-type=application/json&xref_refseq=1&ccds=1&canonical=1&domains=1&hgvs=1&numbers=1&protein=1",
-        "hotspots.url=https://www.cancerhotspots.org/api/hotspots/single/",
-        "hotspots.3d.url=https://www.3dhotspots.org/api/hotspots/3d/",
         "spring.data.mongodb.uri=mongodb://localhost/integration",
         "server.port=38888"
     }
 )
 public class CancerHotspotsIntegrationTest
 {
+    private static List<Hotspot> HOTSPOTS;
+
+    @BeforeClass
+    public static void readMockFile() throws IOException
+    {
+        JsonToObjectMapper objectMapper = new JsonToObjectMapper();
+        // reading the entire DB dump for a full scale test
+        HOTSPOTS = objectMapper.readHotspots("hotspots_db_dump.json");
+    }
+
+
     private RestTemplate restTemplate = new RestTemplate();
 
     @Autowired
-    NotationConverter notationConverter;
+    private NotationConverter notationConverter;
+
+    @Autowired
+    private HotspotRepository hotspotRepository;
+
+    @Before
+    public void setupDB()
+    {
+        // only init once
+        if (this.hotspotRepository.count() == 0) {
+            // save the entire dump into the embedded mongoDB
+            this.hotspotRepository.insert(HOTSPOTS);
```

Comment:
Ideally we should do this in a `@BeforeClass` method, but the repository is not yet autowired in that case, so this is a sort of workaround.